R/arg.R

```diff
@@ -47,30 +48,47 @@ arg_match <- function(arg, values = NULL) {
   }
 
   arg <- arg[[1]]
-  i <- match(arg, values)
+  arg_match0(arg, values, arg_nm)
+}
 
-  if (is_na(i)) {
-    msg <- arg_match_invalid_msg(arg_nm, values)
+#' @description
+#' `arg_match0()` is a bare-bones version if performance is at a premium.
+#' It requires a string as `arg` and explicit `values`.
+#' (For convenience, setting `arg` to `values`, perhaps permutated, is also
+#' supported.)
```

Comment:
For convenience, `arg` may also be a character vector containing every element of `values`, possibly permuted.